src/mscorlib/src/System/Text/UTF8Encoding.cs

```diff
@@ -58,11 +58,13 @@ public class UTF8Encoding : Encoding
         internal sealed class UTF8EncodingSealed : UTF8Encoding
         {
             public UTF8EncodingSealed() : base(encoderShouldEmitUTF8Identifier: true) { }
+            public UTF8EncodingSealed(bool encoderShouldEmitUTF8Identifier) : base(encoderShouldEmitUTF8Identifier) { }
         }
 
         // Used by Encoding.UTF8 for lazy initialization
         // The initialization code will not be run until a static member of the class is referenced
         internal static readonly UTF8EncodingSealed s_default = new UTF8EncodingSealed();
+        internal static readonly UTF8EncodingSealed s_defaultNoBOM = new UTF8EncodingSealed(encoderShouldEmitUTF8Identifier: false);
```

Comment:
This static cache should not be needed. The static cache in Encoding.cs is sufficient.